ckan/controllers/user.py

```diff
@@ -349,6 +349,14 @@ def _save_edit(self, id, context):
 
             email_changed = data_dict['email'] != c.userobj.email
 
+            if id != data_dict['name']:
```

Comment:
Shouldn't this check be done at the validation level? Otherwise the name will be able to be changed via the API